### ISSUE TITLE
Handle namespaces that end in an underscore in turtle serializer

### ISSUE THREAD

**Issue by kshefchek** [RDFLib/rdflib#632]
Consider the following:

```
   from rdflib import Graph, URIRef

   graph = Graph()
   graph.bind('GENO', 'http://purl.obolibrary.org/obo/GENO_')
   graph.bind('RO', 'http://purl.obolibrary.org/obo/RO_')

   graph.add((URIRef('http://example.org'),
                     URIRef('http://purl.obolibrary.org/obo/RO_0002200'),
                     URIRef('http://purl.obolibrary.org/obo/GENO_0000385')))

   print(graph.serialize(format='turtle'))
```

Outputs:

```
    @prefix GENO: <http://purl.obolibrary.org/obo/GENO_> .
    @prefix RO: <http://purl.obolibrary.org/obo/RO_> .
    @prefix ns1: <http://purl.obolibrary.org/obo/> .
    @prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
    @prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
    @prefix xml: <http://www.w3.org/XML/1998/namespace> .
    @prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

   <http://example.org> ns1:RO_0002200 ns1:GENO_0000385 .
```

This appears to be handled in the namespace module: https://github.com/RDFLib/rdflib/blob/master/rdflib/namespace.py#L485

Is this a bug or am I misunderstanding the turtle specs?

**Comment by satra** [comment]
this is also the same issue reported here: https://github.com/trungdong/prov/issues/96

**Comment by kshefchek** [comment]
this looks to be fixed in 5.0 via https://github.com/RDFLib/rdflib/pull/649
